verification/cbmc/proofs/aws_cryptosdk_priv_try_gen_key/aws_cryptosdk_priv_try_gen_key_harness.c

```diff
@@ -0,0 +1,246 @@
+        __CPROVER_assume(aws_string_is_valid(key));
+        __CPROVER_assume(key->len <= UINT8_MAX);
+        val->key                 = key;
+        struct aws_string *value = ensure_string_is_allocated_nondet_length();
+        __CPROVER_assume(aws_string_is_valid(value));
+        __CPROVER_assume(value->len <= UINT8_MAX);
+        val->value = value;
+    }
+}
+
+/* Stub this because of the override of aws_array_list_get_at_ptr
+ Original function is here:
+ https://github.com/aws/aws-encryption-sdk-c/blob/master/source/keyring_trace.c#L235 */
+void aws_cryptosdk_keyring_trace_clear(struct aws_array_list *trace) {
+    AWS_FATAL_PRECONDITION(aws_cryptosdk_keyring_trace_is_valid(trace));
+    AWS_FATAL_PRECONDITION(trace->item_size == sizeof(struct aws_cryptosdk_keyring_trace_record));
+    aws_array_list_clear(trace);
+}
+
+/* Stub this until https://github.com/diffblue/cbmc/issues/5344 is fixed
+ Original function is here:
+ https://github.com/aws/aws-encryption-sdk-c/blob/master/source/edk.c#L44 */
+void aws_cryptosdk_edk_list_clean_up(struct aws_array_list *encrypted_data_keys) {
+    assert(aws_cryptosdk_edk_list_is_valid(encrypted_data_keys));
+    aws_array_list_clean_up(encrypted_data_keys);
+}
+
+/**
+ * Receives encryption request from user and attempts to generate encryption materials,
+ * including an encrypted data key and a list of EDKs for doing encryption.
+ *
+ * On success returns AWS_OP_SUCCESS and allocates encryption materials object at address
+ * pointed to by output.
+ *
+ * On failure returns AWS_OP_ERR, sets address pointed to by output to NULL, and sets
+ * internal AWS error code.
+ */
+int generate_enc_materials(
+    struct aws_cryptosdk_cmm *cmm,
+    struct aws_cryptosdk_enc_materials **output,
+    struct aws_cryptosdk_enc_request *request) {
+    assert(aws_cryptosdk_cmm_base_is_valid(cmm));
+    assert(AWS_OBJECT_PTR_IS_WRITABLE(output));
+    assert(aws_cryptosdk_enc_request_is_valid(request));
+
+    struct aws_cryptosdk_enc_materials *materials = malloc(sizeof(*materials));
+    if (materials == NULL) {
+        *output = NULL;
+        return AWS_OP_ERR;
+    }
+
+    // Set up the allocator
+    // Request->alloc is session->alloc
+    materials->alloc = request->alloc;
+    __CPROVER_assume(aws_allocator_is_valid(materials->alloc));
+    __CPROVER_assume(materials->alloc != NULL);
+
+    // Set up the signctx
+    materials->signctx = ensure_nondet_sig_ctx_has_allocated_members();
+    __CPROVER_assume(aws_cryptosdk_sig_ctx_is_valid_cbmc(materials->signctx));
+
+    // Set up the unencrypted_data_key
+    __CPROVER_assume(aws_byte_buf_is_bounded(&materials->unencrypted_data_key, MAX_BUFFER_SIZE));
+    ensure_byte_buf_has_allocated_buffer_member(&materials->unencrypted_data_key);
+    __CPROVER_assume(aws_byte_buf_is_valid(&materials->unencrypted_data_key));
+
+    // Set up the edk_list
+    // edk_list Precondition: We have a valid list */
```

Comment:
np: I think this comment might've gotten mangled?

Reply:
Yep, fixed!